omnisend/class-omnisend-core-bootstrap.php

```diff
@@ -182,17 +176,22 @@ public static function cron_schedules( $schedules ) {
 
 
 	private static function migrate_options() {
-		$is_store_connected = Options::is_store_connected();
-		$visit_count        = Options::get_landing_page_visit_count();
+		$landing_page_visited            = Options::is_landing_page_visited();
+		$landing_page_notification_state = Options::get_landing_page_notification_state();
+		$landing_page_last_visit_time    = Options::get_landing_page_last_visit_time();
 
-		$store_connected_before_visit_count_option = $is_store_connected && $visit_count === 0;
-		if ( $store_connected_before_visit_count_option ) {
+		if ( $landing_page_visited && ( $landing_page_notification_state === NOTIFICATION_NOT_SHOWN || $landing_page_last_visit_time === 0 ) ) {
 			Options::set_landing_page_visited();
```

Comment:
this seems illogical, if landing_page_visited then call set_landing_page_visited

Reply:
read whole condition